binder/environment.yml

```diff
@@ -2,11 +2,14 @@ name: binder
 channels:
   - conda-forge
 dependencies:
-  - python=3.9
-  - jupyter-server-proxy>=1.4
+  - python=3.11
```

Comment:
Hi @psobolewskiPhD :smile: Is there any technical reason not to bump to 3.12?

Reply:
No particular reason. We support 3.9-3.12 and suggest 3.10 in the install instructions:
https://napari.org/stable/tutorials/fundamentals/installation.html#install-as-python-package-recommended
I just chose 3.11 because wanted to get the better error messages, which are nice for teaching.

Reply:
3.12 would give you more performance than 3.11. 3.11 is fine though until there is time to check the dependencies with 3.12.

Reply:
Yeah, we've been quite conservative I think--I personally use Qt6 and python 3.12.
I will make an issue to keep track of this. We can look at the plugins suggested/used in the notebooks and ensure there are no issues with 3.12.